src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -639,31 +633,30 @@ private bool AcquireClientCredentials(ref byte[]? thumbPrint, bool newCredential
                         NetEventSource.Log.UsingCachedCredential(this);
                     _credentialsHandle = cachedCredentialHandle;
                     cachedCred = true;
-                    if (selectedCert != null)
-                    {
-                        _sslAuthenticationOptions.CertificateContext ??= SslStreamCertificateContext.Create(selectedCert!);
-                    }
+                    UpdateCertificateContext(selectedCert);
```

Comment:
This always sets `OwnsCertificateContext = true`. Do we always have copy of the user provided cert at this point? 
And I'm wondering if we should add test verifying that when user gives us server or client certificate  they are not disposed after SslStream is.

Reply:
We do not always have a copy. This PR does not dispose the TargetCertificate as the lifetime of that is (has always been) owned by the user.